app.rb

```diff
@@ -47,84 +52,76 @@ def operation(input)
 
   def list_books
     puts '-' * 50
-    books = Book.class_variable_get(:@@books)
-    if books.empty?
+    if @books.empty?
       puts 'The books list is empty'
     else
       puts 'Books list:'
-      books.each_with_index do |book, index|
-        puts "[Book #{index + 1}] Title: #{book.title} | Author: #{book.author}"
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}-[Book] Title: #{book.title} | Author: #{book.author}"
       end
     end
   end
 
   def list_people
     puts '-' * 50
-    people = Person.class_variable_get(:@@people)
-    if people.empty?
-      puts 'The list is empty'
+    if @people.empty?
+      puts "The people\'s list is empty"
     else
       puts 'People list:'
-      people.each_with_index do |person, index|
-        puts "[Teacher #{index + 1}] ID: #{person.id} | Name: #{person.name} | Age: #{person.age}" if person.is_a?(Teacher)
-        puts "[Student #{index + 1}] ID: #{person.id} | Name: #{person.name} | Age: #{person.age}" if person.is_a?(Student)
+      @people.each_with_index do |per, i|
+        puts "#{i + 1}-[Teacher] ID: #{per.id} | Name: #{per.name} | Age: #{per.age}" if per.is_a?(Teacher)
+        puts "#{i + 1}-[Student] ID: #{per.id} | Name: #{per.name} | Age: #{per.age}" if per.is_a?(Student)
       end
     end
   end
 
   def create_person
     puts '-' * 50
-    puts 'Do you want to create a student (1) or a teacher (2)? [input the number]: '
-    input = gets.chomp.to_i
-    case input
+    print 'Do you want to create a student (1) or a teacher (2)? [input the number]: '
+    option = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    case option
     when 1
-      create_student
+      create_student(name, age)
     when 2
-      create_teacher
+      create_teacher(name, age)
     else
-      puts 'Please input a valid number'
+      puts 'Please input a valid number (1 or 2)'
     end
   end
 
-  def create_student
-    puts "Student\'s name : "
-    name = gets.chomp
-    puts "Student\'s age: "
-    age = gets.chomp
-    puts "Has parent\'s persmission? [Y/N]: "
+  def create_student(name, age)
+    print "Has parent\'s persmission? [Y/N]: "
     parent_permission = gets.chomp
     parent_permission = true if parent_permission == ('y' || 'Y')
     parent_permission = false if parent_permission == ('n' || 'N')
     Student.new(age, 'class', name, parent_permission: parent_permission)
     puts "Student (#{name}) has been created successfully"
   end
 
-  def create_teacher
-    puts "Teacher\'s name : "
-    name = gets.chomp
-    puts "Teacher\'s age: "
-    age = gets.chomp
-    puts "Teacher\'s specialization: "
+  def create_teacher(name, age)
+    print "Teacher\'s specialization: "
```

Comment:
- [ ] in this method you are printing to the console, getting user response and processing the response, this is not in line with the single responsibility rule, please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.